components/CompanyRegister.tsx

```diff
@@ -1,42 +1,72 @@
-import { CreateCompanyItem } from '@/db/entities/company';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useForm } from 'react-hook-form';
+import { useAccount } from 'wagmi';
 import useCompany from '@/hooks/useCompanyRegistration';
-import { isSuccessResponse } from '@/pages/api/companies';
+import { useRecruitmentRegisterCompany } from '../src/generated';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 import { toast } from 'react-toastify';
-import { useAccount } from 'wagmi';
-import { useRecruitmentRegisterCompany } from '../src/generated';
+import { isSuccessResponse } from '@/pages/api/companies';
 import { Badge } from './ui/badge';
 
+const schema = yup.object().shape({
```

Comment:
lets switch to zod.  It has better type safety:

https://www.npmjs.com/package/@hookform/resolvers#zod

Reply:
noted, zod dependency has been installed and used